packages/legend-application-studio/src/components/editor/edit-panel/uml-editor/TaggedValueEditor.tsx

```diff
@@ -101,99 +133,364 @@ export const TaggedValueEditor = observer(
     const [isExpanded, setIsExpanded] = useState(false);
     const toggleExpandedMode = (): void => setIsExpanded(!isExpanded);
 
+    // Drag and Drop
+    const handleHover = useCallback(
+      (item: AllTaggedValueDragSource, monitor: DropTargetMonitor): void => {
+        const draggingProperty = item.taggedValue;
+        const hoveredProperty = taggedValue;
+        if (_whoKnows instanceof Class) {
+          class_arrangeTaggedValues(
+            _whoKnows,
+            draggingProperty,
+            hoveredProperty,
+          );
+        } else if (_whoKnows instanceof Enum) {
+          enum_arrangeTaggedValues(
+            _whoKnows,
+            draggingProperty,
+            hoveredProperty,
+          );
+        } else if (_whoKnows instanceof Enumeration) {
+          enumeration_arrangeTaggedValues(
+            _whoKnows,
+            draggingProperty,
+            hoveredProperty,
+          );
+        } else if (_whoKnows instanceof Association) {
+          association_arrangeTaggedValues(
+            _whoKnows,
+            draggingProperty,
+            hoveredProperty,
+          );
+        } else if (_whoKnows instanceof ConcreteFunctionDefinition) {
+          function_arrangeTaggedValues(
+            _whoKnows,
+            draggingProperty,
+            hoveredProperty,
+          );
+        } else if (_whoKnows instanceof DataElement) {
+          dataElement_arrangeTaggedValues(
+            _whoKnows,
+            draggingProperty,
+            hoveredProperty,
+          );
+        } else if (_whoKnows instanceof Property) {
+          property_arrangeTaggedValues(
+            _whoKnows,
+            draggingProperty,
+            hoveredProperty,
+          );
+        } else {
+          //taggedvalueeditor parent class is something else
+        }
+      },
+      [_whoKnows, taggedValue],
+    );
+
+    const [{ isBeingDraggedTaggedValue }, dropConnector] = useDrop(
+      () => ({
+        accept: [CLASS_TAGGED_VALUE_DND_TYPE.TAGGED_VALUE],
+        hover: (
+          item: AllTaggedValueDragSource,
+          monitor: DropTargetMonitor,
+        ): void => handleHover(item, monitor),
+        collect: (
+          monitor,
+        ): { isBeingDraggedTaggedValue: TaggedValue | undefined } => ({
+          isBeingDraggedTaggedValue:
+            monitor.getItem<AllTaggedValueDragSource | null>()?.taggedValue,
+        }),
+      }),
+      [handleHover],
+    );
+    const isBeingDragged =
+      projectionTaggedValueState.taggedValue === isBeingDraggedTaggedValue;
+
+    const [, dragConnector, dragPreviewConnector] = useDrag(
+      () => ({
+        type: CLASS_TAGGED_VALUE_DND_TYPE.TAGGED_VALUE,
+        item: (): AllTaggedValueDragSource => {
+          projectionTaggedValueState.setIsBeingDragged(true);
```

Comment:
I thought we talked about the fact that we don't need these states at all?

Reply:
I think we got rid of propertyStateTesting: Property[] in ClassState for example but that was before the handler of useDrop and dragDrop changes which projectionPropertyState in ClassEditor (and other projections) uses now because of the UI changing so I think these states are now attached and necessary for the isBeingDragged to render conditionally ^^;